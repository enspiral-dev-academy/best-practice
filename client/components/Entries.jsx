import React, { useEffect } from "react";
import { Link } from "react-router-dom";

import WrappedEntryItem from "./EntryItem";
import WrappedWaitIndicator from "./WaitIndicator"

import { wrappedWith, authentication, entriesContext } from "../wrappers";

export function Entries({ authenticated, entries, retrieveEntries }) {
  useEffect(() => {
    retrieveEntries();
  }, []);
  return (
    <>
      <h2>Entries</h2>
      {authenticated() && <Link to="/add">Add an entry</Link>}
      <WrappedWaitIndicator />
      <ul>
        {entries.map((entry) => (
          <WrappedEntryItem key={entry.id} entryData={entry} />
        ))}
      </ul>
    </>
  );
}

export default wrappedWith(authentication, entriesContext)(Entries);
