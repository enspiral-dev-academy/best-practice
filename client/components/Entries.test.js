import React from "react";
import { renderWithRouter } from "../testing/utils";
import "@testing-library/jest-dom";

import { Entries } from "./Entries";

import mockEntries from "../testing/mockEntries";

test("renders entries page correctly when authenticated", () => {
  const { asFragment } = renderWithRouter(
    <Entries
      authenticated={() => true}
      entries={mockEntries}
      retrieveEntries={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders entries page correctly when not authenticated", () => {
  const { asFragment } = renderWithRouter(
    <Entries
      authenticated={() => false}
      retrieveEntries={() => {}}
      entries={mockEntries}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("useEffect calls retrieveEntries on mount", () => {
  const retrieveEntries = jest.fn();
  renderWithRouter(
    <Entries
      authenticated={() => false}
      entries={mockEntries}
      retrieveEntries={retrieveEntries}
    />
  );

  expect(retrieveEntries).toHaveBeenCalled();
});
