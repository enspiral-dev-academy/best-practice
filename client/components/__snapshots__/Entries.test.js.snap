// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders entries page correctly when authenticated 1`] = `
<DocumentFragment>
  <h2>
    Entries
  </h2>
  <a
    href="/add"
  >
    Add an entry
  </a>
  <ul>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/1"
      >
        mocked entry 1
      </a>
    </li>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/2"
      >
        mocked entry 2
      </a>
    </li>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/3"
      >
        mocked entry 3
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`renders entries page correctly when not authenticated 1`] = `
<DocumentFragment>
  <h2>
    Entries
  </h2>
  <ul>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/1"
      >
        mocked entry 1
      </a>
    </li>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/2"
      >
        mocked entry 2
      </a>
    </li>
    <li
      data-testid="entry"
    >
      <a
        href="/entry/3"
      >
        mocked entry 3
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
